pyanaconda/modules/subscription/system_purpose.py

```diff
@@ -137,3 +139,67 @@ def process_field(syspurpose_value, valid_values, value_name):
         return syspurpose_value
     else:
         return ""
+
+
+def give_the_system_purpose(sysroot, role, sla, usage, addons):
+    """Set system purpose for the installed system by calling the syspurpose tool.
+
+    The tool is called in the installed system chroot, so this method can be only
+    called once the system rootfs content is in place.
+
+    :param role: role of the system
+    :type role: str or None
+    :param sla: Service Level Agreement for the system
+    :type sla: str or None
+    :param usage: intended usage of the system
+    :type usage: str or None
+    :param list addons: any additional layered products or features
+    """
+
+    if role or sla or usage or addons:
+        # using join_paths() as both paths are absolute
+        syspurpose_sysroot_path = util.join_paths(sysroot, SYSPURPOSE_UTILITY_PATH)
+        if os.path.exists(syspurpose_sysroot_path):
+            # The syspurpose utility can only set one value at a time,
+            # so we might need to call it multiple times to set all the
+            # requested values.
+            #
+            # Also as the values can contain white space we need to make sure the
+            # values passed to arguments are all properly quoted.
+            if role:
+                args = ["set-role", '{}'.format(role)]
+                rc = util.execWithRedirect("syspurpose", args, root=sysroot)
+                if rc:
+                    log.error("subscription: syspurpose failed to set role, rc: %s", rc)
+                    return False
+
+            if sla:
+                args = ["set-sla", '{}'.format(sla)]
+                rc = util.execWithRedirect("syspurpose", args, root=sysroot)
+                if rc:
+                    log.error("subscription: syspurpose failed to set sla, rc: %s", rc)
+                    return False
+
+            if usage:
+                args = ["set-usage", '{}'.format(usage)]
```

Comment:
I don't understand the formatting. It could be just `usage` instead of `'{}'.format(usage)`, no? Same for `role`, `sla` and `addons`.

Reply:
I did some digging and its due to quoting that was later dropped: 66ac3f343eed3cabfba75cc24e54b49f0c7307fb
So I think we can indeed drop that `format()` call. :)